substrate/frame/contracts/src/wasm/mod.rs

```diff
@@ -201,19 +206,14 @@ impl<T: Config> WasmBlob<T> {
 	/// When validating we pass `()` as `host_state`. Please note that such a dummy instance must
 	/// **never** be called/executed, since it will panic the executor.
 	pub fn instantiate<E, H>(
```

Comment:
For enabling skipping of Wasm validation what I would have expected is to add another parameter to this function, e.g. `validate: bool`. And based on `validate` you'd either call `LoadedModule::new` or `LoadedModule::new_unchecked`. For `upload_code` we'd set `validate` to `true` and for execution we'd set `validate` to `false`. This should require the minimum amount of changes needed to introduce this feature.

Reply:
I splitted the responsability, between validate and instantiate.
This also helped spotting a potential bug where we were loading (unnecessarily) twice the module for validation 

so tldr we build and pass an unchecked module when we execute the code, we pass a checked module when we just want to validate

Reply:
I will cherry pick these changes on a clean PR and rebase this one on top so we can review that separately